spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/ConsumerProperties.java

```diff
@@ -209,4 +220,12 @@ public boolean isUseNativeDecoding() {
 	public void setUseNativeDecoding(boolean useNativeDecoding) {
 		this.useNativeDecoding = useNativeDecoding;
 	}
+
+	public boolean isNativeMultipleBinding() {
+		return nativeMultipleBinding;
+	}
+
+	public void setNativeMultipleBinding(boolean nativeMultipleBinding) {
+		this.nativeMultipleBinding = nativeMultipleBinding;
+	}
```

Comment:
@sobychacko and I just had a quick chat about this. Currently the `native-multiple-binding` will be a public property visible through content assist and essentially exposed to the end user. Yet the intention for this property are strictly internal. 
So we agreed that it would probably be wise to remove both _setter/getter_ and in KafkaStreams binder (where it is now required) simply use `DirectFieldAccessor` to set the value and in `BindingService` to access the value.

Not ideal, but will allows us a simple and quick solution for the time being until we get a chance to refactor the code to ensure that we can introduce internal vs external properties properly.

Reply:
>Yet the intention for this property are strictly internal.

I disagree - with the rabbit/kafka binders, the user should be able to choose whether to use a container per destination (as currently) or multiple destinations per container.

While I am not suggesting we can do this, but with rabbit/kafka we could even support

    destination=[foo,bar],[baz,qux]

(2 containers with 2 destinations each).

Reply:
Gary, I can't say I am disagreeing with you, but unless we support this functionality already I do not want to make it public and base on my discussion with Soby, KafkaStreams is the only binder where it always has to be `true`.

Reply:
>but unless we support this functionality already

I thought that was the whole point, i.e. we __do__ intend to support it in the other binders in 2.1 and this is the preparation for that.

Reply:
This specific property and its name basically tells the Binder to "stop being clever" and pass the value to the underlying system exactly as provided. Nothing more.
What you're talking about is functionality for the binder to get "more clever" and I think these are two completely orthogonal concerns - _delegate to the target system_ OR _do something "clever" and here is another property that will help you to define what "clever" means_.